util/src/main/java/io/kubernetes/client/PortForward.java

```diff
@@ -0,0 +1,180 @@
+        this.apiClient = apiClient;
+    }
+
+    private String makePath(String namespace, String name) {
+          return "/api/v1/namespaces/" +
+            namespace +
+            "/pods/" +
+            name +
+            "/portforward";
+    }
+
+    /**
+     * PortForward to a container
+     * 
+     * @param pod The pod where the port forward is run.
+     * @param ports The ports to forward
+     */
+    public PortForwardResult forward(V1Pod pod, List<Integer> ports) throws ApiException, IOException {
+        return forward(pod.getMetadata().getNamespace(), pod.getMetadata().getNamespace(), ports);
+    }
+
+    /**
+     * PortForward to a container.
+     * 
+     * @param namespace The namespace of the Pod
+     * @param name The name of the Pod
+     * @param ports The ports to forward
+     */
+    public PortForwardResult forward(String namespace, String name, List<Integer> ports) throws ApiException, IOException {
+        String path = makePath(namespace, name);
+        WebSocketStreamHandler handler = new WebSocketStreamHandler();
+        PortForwardResult result = new PortForwardResult(handler, ports);
+        List<Pair> queryParams = new ArrayList<>();
+        queryParams.add(new Pair("ports", "80"));
+        WebSockets.stream(path, "GET", queryParams, apiClient, handler);
+        
+        // Wait for streams to start.
+        result.init();
+
+        return result;
+    }
+
+    /**
+     * PortForwardResult contains the result of an Attach call, it includes streams for stdout
+     * stderr and stdin. 
+     */
+    public static class PortForwardResult { 
+        private WebSocketStreamHandler handler;
+        private HashMap<Integer, Integer> streams;
+        private List<Integer> ports;
+
+        public PortForwardResult(WebSocketStreamHandler handler, List<Integer> ports) throws IOException {
+            this.handler = handler;
+            this.streams = new HashMap<>();
+            this.ports = ports;
+        }
+
+        public void init() throws IOException {
+            for (int i = 0; i < ports.size(); i++) {
+                InputStream is = handler.getInputStream(i);
+                byte[] data = new byte[2];
+                is.read(data);
+                int port = data[0] + data[1] * 256;
+                streams.put(port, i);
+            }
+        }
+
+        private int findPortIndex(int portNumber) {
+            Integer ix = streams.get(portNumber);
+            if (ix == null) {
+                return -1;
+            }
+            return ix.intValue();
+        }
+
+        /**
+         * Get the output stream for the specified port number (e.g. 80)
+         * @param port The port number to get the stream for.
+         */
+        public OutputStream getOutboundStream(int port) {
+            int portIndex = findPortIndex(port);
+            if (portIndex == -1) {
+                throw new IllegalArgumentException("No such port!");
```

Comment:
Would it be neater to have `findPortIndex(..)` throw this exception instead?

Reply:
switched to return null instead of throw.